libbeat/autodiscover/providers/kubernetes/kubernetes.go

```diff
@@ -145,28 +146,23 @@ func (p *Provider) emitEvents(pod *kubernetes.Pod, flag string, containers []*ku
 	host := pod.Status.GetPodIP()
 
 	// Do not emit events without host (container is still being configured)
-	if host == "" {
+	if host == "" && flag != "stop" {
 		return
 	}
 
-	// Collect all container IDs and runtimes from status information.
-	containerIDs := map[string]string{}
+	// Collect all runtimes from status information.
 	runtimes := map[string]string{}
 	for _, c := range containerstatuses {
-		cid, runtime := kubernetes.ContainerIDWithRuntime(c)
-		containerIDs[c.GetName()] = cid
+		_, runtime := kubernetes.ContainerIDWithRuntime(c)
 		runtimes[c.GetName()] = runtime
 	}
 
 	// Emit container and port information
 	for _, c := range containers {
-		cid := containerIDs[c.GetName()]
+		// This must be an id that doesn't depend on the state of the container
+		// so it works also on `stop` if containers have been already stopped.
+		cid := fmt.Sprintf("%s.%s", pod.Metadata.GetUid(), c.GetName())
 
-		// If there is a container ID that is empty then ignore it. It either means that the container is still starting
-		// up or the container is shutting down.
-		if cid == "" {
-			continue
```

Comment:
what about the case where the container is still starting?

Reply:
I was thinking that just checking for the ID doesn't ensure that the container is ready, so it wouldn't matter so much. But the truth is that there is no need to change this here. I will keep the check for non-stop events.

Thanks for taking the time to review this!